mesosphere/kudo-cassandra-operator
# Implement Readiness Probe

## What changes are proposed in this PR? 
Resolves  [DCOS-58972](https://jira.mesosphere.com/browse/DCOS-58972) 

This PR Implement Readiness Probe for Kudo Cassandra operator. 
LivenessProbe wasn't implemented because we need clear understand when pods definitely must be [restarted](https://cloud.google.com/blog/products/gcp/kubernetes-best-practices-setting-up-health-checks-with-readiness-and-liveness-probes).

Implementation steps : 
1. Added k8s ReadinessProbe properties to `params.yml`
2. Modified `stateful-set.yaml`
3. Added ReadinessProbe script similar like in [`DC/OS`](https://github.com/mesosphere/dcos-commons/blob/master/frameworks/cassandra/src/main/dist/svc.yml#L154-L168) but without security support (which should be implemented and script modified) 
## How were these changes tested?
Manually tested  
1. Install operator 
2. Look into pod events 
`kubectl get event --namespace $kudo_cassandra_instance_namespace --field-selector involvedObject.name=cassandra-node-0` 
and
`kubectl describe pod cassandra-node-0`
3. Check that all pods were run and healthy.
